File: 01-Assignment-02-02-2026/Using_C/program_11.c
/* Program 11: Find the geometric mean of n numbers */

/* Program 11: Find the geometric mean of n numbers */

#include <stdio.h>
#include <math.h>

int main() {
    int n;
    printf("Total Numbers : ");
    scanf("%d", &n);

    double product = 1.0, temp;

    for (int i = 0; i < n; i++) {
        printf("Enter Number x%d : ", i + 1);
        scanf("%lf", &temp);
        product *= temp;
    }

    double gm = pow(product, 1.0 / n);
    printf("Geometric Mean of Given Numbers is %.2f", gm);

    return 0;
}

File: 01-Assignment-02-02-2026/Using_Java/Program11.java
/*
Program 11
Java program to find the geometric mean of n numbers
*/

import java.util.Scanner;

public class Program11 {

    public static void main(String[] args) {
        // GP = geometric mean => if a,b,c are in GP
        // then 3rd root of (a*b*c) is GM

        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter the total number of values: ");
        int size = scanner.nextInt();

        if (size <= 0) {
            System.out.println("Size must be greater than 0.");
            scanner.close();
            return;
        }

        double product = 1;

        for (int i = 0; i < size; i++) {
            System.out.print("Enter number " + (i + 1) + ": ");
            double num = scanner.nextDouble();
            product *= num;
        }

        double geometricMean = Math.pow(product, 1.0 / size);

        System.out.println("Geometric Mean = " + geometricMean);

        scanner.close();
    }
}

